### ISSUE TITLE
Retry does not work for version 10 of the client.

### ISSUE THREAD

**Issue by JFreakDK** [okta/okta-sdk-java#842]
### Describe the bug?

It is possible to configure retry limits and connection timeouts like this:
```
        ApiClient client = Clients.builder()
                .setRetryMaxAttempts(4)
                .setRetryMaxElapsed(10)
                .setConnectionTimeout(2)
```
No retries seem to be executed.

### What is expected to happen?

If a request times out or fails before the `retryMaxElapsed` limit has been reached the request should be executed again.
The same is expected if a request fails with a http error before the `retryMaxElapsed` limit has been reached.

### What is the actual behavior?

The client throws an exception after the first request times out.
```
org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://<hostname>/api/v1/users/username": Read timed out; nested exception is java.net.SocketTimeoutException: Read timed out

	at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:785)
	at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:670)
	at org.openapitools.client.ApiClient.lambda$invokeAPI$2(ApiClient.java:713)
	at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)
	at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)
	at org.openapitools.client.ApiClient.invokeAPI(ApiClient.java:713)
	at org.openapitools.client.api.UserApi.getUserWithHttpInfo(UserApi.java:1921)
	at org.openapitools.client.api.UserApi.getUser(UserApi.java:1875)
```

### Reproduction Steps?

The test case here reproduce the issue: https://github.com/JFreakDK/okta-retry-test

Feel free to add the test to your test suite.

### Java Version

openjdk 19.0.2 2023-01-17
OpenJDK Runtime Environment Homebrew (build 19.0.2)
OpenJDK 64-Bit Server VM Homebrew (build 19.0.2, mixed mode, sharing)

### SDK Version

10.2.2

**Comment by arvindkrishnakumar-okta** [comment]
@JFreakDK Thanks for posting!

The SDK is designed to automatically retry on HTTP 529 errors only (See https://github.com/okta/okta-sdk-java/blob/master/api/src/main/resources/custom_templates/ApiClient.mustache#L944-L965 & https://github.com/okta/okta-sdk-java/blob/master/api/src/main/java/com/okta/sdk/error/ErrorHandler.java#L114-L117). Also, retry is disabled if both `okta.client.requestTimeout ` &  `okta.client.rateLimit.maxRetries` config variables are set to zero).

**Comment by JFreakDK** [comment]
Interesting. Thank you for the reply. According to our tests that is a change in behavior from 8.x then. Maybe you should add that to the migration guide.

**Comment by sahilnarwal-nxt** [comment]
> @JFreakDK Thanks for posting!
> 
> The SDK is designed to automatically retry on HTTP 529 errors only (See https://github.com/okta/okta-sdk-java/blob/master/api/src/main/resources/custom_templates/ApiClient.mustache#L944-L965 & https://github.com/okta/okta-sdk-java/blob/master/api/src/main/java/com/okta/sdk/error/ErrorHandler.java#L114-L117). Also, retry is disabled if both `okta.client.requestTimeout ` & `okta.client.rateLimit.maxRetries` config variables are set to zero).

As per sdk doc this looks more like a bug
https://github.com/okta/okta-sdk-java#connection-retry
`By default, this SDK will retry requests that return with response code 503, 504, 429(caused by rate limiting), or socket/connection exceptions.`

**Comment by arvindkrishnakumar-okta** [comment]
This should be fixed with 11.0.0 

Feel free to post followup comments, if necessary.
